### PR TITLE
Arcade bootstrapper: msbuildEngine, InitializeBuildTool

### PR DESCRIPTION
**Pull request by tmat**
Use more of Arcade bootstrapper functions.
Convert `buildCoreClr` to `msbuildEngine` parameter used by Arcade.
Update Initialize* functions to memoize their result.

### CODE REVIEW

**Comment by tmat** [comment]
@jaredpar PTAL

**Comment by tmat** [comment]
@agocke @333fred @dotnet/roslyn-infrastructure

File: build/scripts/build.ps1 [hunk 1]
```diff
@@ -340,8 +329,8 @@ function Test-XUnitCoreClr() {
 # Core function for running our unit / integration tests tests
 function Test-XUnit() {
 
-    # Used by tests to locate dotnet CLI
-    $env:DOTNET_INSTALL_DIR = Split-Path $dotnet -Parent
+    # Tests need to locate .NET Core SDK
+    InitializeDotNetCli
```
Comment (jaredpar) [inline build/scripts/build.ps1]: Where is this defined?

Comment (tmat) [reply build/scripts/build.ps1]: Damn PowerShell :(

**Comment by tmat** [comment]
Superseded by https://github.com/dotnet/roslyn/pull/31511

**Pull request closed without merging**